### DISCUSSION TITLE
Language server does not start when configured with Nvim LSP

### DISCUSSION THREAD

**Discussion by DanielPower** [polarmutex/beancount-language-server#160] [Q&A] [answered]
I've tried the following installations:

- 1.0.2 from Cargo
- 1.0.2 from Arch User Repository
- 1.0.3 installed via nvim-lsp-installer (by modifying their install script, since their parser hasn't been updated for the new file name structure in 1.0.3).

LspInfo shows that the file I have open is of filetype: beancount. And it shows that the language server is executable, and set to autostart. But it does not start. Additionally, if I run `LspStart beancount` manually, nothing happens.

![1651453686](https://user-images.githubusercontent.com/12166320/166173113-33a65091-8c49-4016-b98f-ebe698635024.png)

I'm not aware of any logs that might have any relevant information, but would appreciate it if you have any advice.

Attached is the config I'm passing to lspconfig.

![1651453850](https://user-images.githubusercontent.com/12166320/166173168-5ab90aa7-3414-49b2-9814-25df6313173c.png)

I'm not sure what the `journalFile` is supposed to refer to. So that could be the problem, since mine is blank. If that needs to be populated, what should I point it to?

**Comment by polarmutex** [comment]
journal_file is the root or initial file for your beancount journal. Basically the file you pass into the beancount Cli 

There is no way that I know to find all definitions of accounts and payees for completions from opening a beancount file(I split my transactions in many files). So I need
To improve the error handling of this case. But for many they have just one file so I should use the current open file if none exists. I will look into improving this

**Reply by DanielPower** [reply]
Oh, that makes sense. In that case I do think it would make sense to use the current file if none is provided, since many people (myself included) use a single massive ledger file.

I did actually try setting `journalFile` to the absolute path to my ledger, and that didn't work. Perhaps it needs to be a path relative to the root? I'll try that later today.

Thanks for your quick response! I'm new to Beancount. Just migrated my Gnucash over a couple days ago. Plaintext accounting is very exciting, and I appreciate your work to streamline the experience!

Edit: And once I get it working, I'll put up a PR against nvim-lsp-installer to fix the downloading, if they haven't already done so by then :)

**Comment by polarmutex** [comment, marked as answer]
I forgot that in the rust rewrite I renamed it to "journal_file". try that, I just prefer snake case to camel

**Reply by DanielPower** [reply]
Ah, that's almost definitely it. I'll try it when I get home.

**Reply by DanielPower** [reply]
That did it. Thank you!

**Reply by dustinfarris** [reply]
is there a replacement for `pythonPath` as well?
